chemprop/v2/data/utils.py

```diff
@@ -2,65 +2,253 @@
 from typing import Sequence
 
 import numpy as np
-from rdkit.Chem.Scaffolds import MurckoScaffold
 
 
 from chemprop.v2.data.datapoints import MoleculeDatapoint
+from chemprop.v2.data.datasets import MoleculeDataset
+
+from astartes import train_val_test_split, train_test_split
+from astartes.molecules import train_val_test_split_molecules, train_test_split_molecules
+
+import copy
+from logging import Logger
+import pickle
+from random import Random
+from typing import Tuple
+import os
+
+from chemprop.data import log_scaffold_stats
+from chemprop.args import TrainArgs
 
 
 def split_data(
     data: Sequence[MoleculeDatapoint],
     split: str = "random",
-    sizes: tuple[float, float, float] = (0.8, 0.1, 0.1),
-    k: int = 5,
-    fold: int = 0,
-):
+    sizes: Tuple[float, float, float] = (0.8, 0.1, 0.1),
+    key_molecule_index: int = 0,
+    seed: int = 0,
+    num_folds: int = 1,
+    args: TrainArgs = None,
+    logger: Logger = None,
+) -> Tuple[MoleculeDataset, MoleculeDataset, MoleculeDataset]:
+    r"""
+    Splits data into training, validation, and test splits.
+
+    :param data: A :class:`~chemprop.data.MoleculeDataset`.
+    :param split_type: Split type.
+    :param sizes: A length-3 tuple with the proportions of data in the train, validation, and test sets.
+    :param key_molecule_index: For data with multiple molecules, this sets which molecule will be considered during splitting.
+    :param seed: The random seed to use before shuffling data.
+    :param num_folds: Number of folds to create (only needed for "cv" split type).
+    :param args: A :class:`~chemprop.args.TrainArgs` object.
+    :param logger: A logger for recording output.
+    :return: A tuple of :class:`~chemprop.data.MoleculeDataset`\ s containing the train,
+             validation, and test splits of the data.
+    """
     if not (len(sizes) == 3 and np.isclose(sum(sizes), 1)):
         raise ValueError(f"Invalid train/val/test splits! got: {sizes}")
 
-    n_train, n_val, n_test = [int(size * len(data)) for size in sizes]
-
-    if split == "random":
-        idxs = np.arange(len(data))
-        np.random.shuffle(idxs)
-
-        train = [data[i] for i in idxs[:n_train]]
-        val = [data[i] for i in idxs[n_train : n_train + n_val]]
-        test = [data[i] for i in idxs[n_train + n_val :]]
-    elif split == "scaffold":
-        scaffold2idxs = defaultdict(set)
-        for i, d in enumerate(data):
-            scaffold = MurckoScaffold.MurckoScaffoldSmiles(mol=d.mol)
-            scaffold2idxs[scaffold].add(i)
-
-        big_index_sets = []
-        small_index_sets = []
-        for idxs in scaffold2idxs.values():
-            if len(idxs) > n_val / 2 or len(idxs) > n_test / 2:
-                big_index_sets.append(idxs)
+    random = Random(seed)
```

Comment:
Is this object 'random' used? If so, I feel it could be moved closer to where it is used.

Reply:
this is (was) still used by some of the samplers i had not yet moved to `astartes`. I have made the changes locally and will push them later - just finishing porting the repeated smiles sampler

Reply:
Done